### PR TITLE
refactor: removing unneccesaary l2_to_l1 calls in account_faulty

### PR DESCRIPTION
**Pull request by aner-starkware**
This change is [Reviewable](https://reviewable.io/reviews/starkware-libs/blockifier/1542)

### CODE REVIEW

**Review by dorimedini-starkware — CHANGES_REQUESTED** [review]
> Reviewed 5 of 5 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/blockifier/1542)* status: all files reviewed, 2 unresolved discussions (waiting on @aner-starkware, @elintul, and @MohammadNassar1)

*[a discussion](https://reviewable.io/reviews/starkware-libs/blockifier/1542#-Nqws9AvBvFEig6-gdKQ:-Nqws9AvBvFEig6-gdKR:b-6m9k8m) (no related file):*
@elintul or @MohammadNassar1 need your approval here; this code was added by mohammad in [PR 606](https://reviewable.io/reviews/starkware-libs/blockifier/606) (related to [601](https://reviewable.io/reviews/starkware-libs/blockifier/601)... somehow)

___
___
*[`crates/blockifier/feature_contracts/cairo1/account_faulty.cairo` line 0 at r1](https://reviewable.io/reviews/starkware-libs/blockifier/1542#-Nqwrut1Ap_O4Ajt0uJy:-Nqwrut1Ap_O4Ajt0uJz:b-la3ujb) ([raw file](https://github.com/starkware-libs/blockifier/blob/78ac5d1365d5ec23c26c10a2472fa7e4cdb83034/crates/blockifier/feature_contracts/cairo1/account_faulty.cairo#L0)):*
there are many unrelated formatting changes here; why?
(you may be the only one with properly configured Cairo formatter lol)

**Review by elintul — CHANGES_REQUESTED** [review]
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/blockifier/1542)* status: all files reviewed, 3 unresolved discussions (waiting on @aner-starkware and @MohammadNassar1)

*[a discussion](https://reviewable.io/reviews/starkware-libs/blockifier/1542#-Nqwtg1F9EP6FoTBE2Yc:-Nqwtg1G1KQi0MBrHKvY:bhb0srs) (no related file):*
Why are they unnecessary? 

___

**Review by dorimedini-starkware — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/blockifier/1542)* status: all files reviewed, 3 unresolved discussions (waiting on @aner-starkware, @elintul, and @MohammadNassar1)

*[a discussion](https://reviewable.io/reviews/starkware-libs/blockifier/1542#-Nqwtg1F9EP6FoTBE2Yc:-NqwuBC56FzJVI1QkGWN:bg4ft39) (no related file):*
<details><summary><i>Previously, elintul (Elin) wrote…</i></summary><blockquote>

Why are they unnecessary?
</blockquote></details>

from the comments in PR 606 it seems like it's just for debugging 601.
Also, no tests fail when these are removed... seems suspicious

___

**Review by dorimedini-starkware — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/blockifier/1542)* status: all files reviewed, 3 unresolved discussions (waiting on @aner-starkware, @elintul, and @MohammadNassar1)

*[a discussion](https://reviewable.io/reviews/starkware-libs/blockifier/1542#-Nqwtg1F9EP6FoTBE2Yc:-NqwuJvbCYUdzeiEk6f2:bvpco3k) (no related file):*
<details><summary><i>Previously, dorimedini-starkware wrote…</i></summary><blockquote>

from the comments in PR 606 it seems like it's just for debugging 601.
Also, no tests fail when these are removed... seems suspicious
</blockquote></details>

why were they added? which test needs them?

___

**Review by aner-starkware — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/blockifier/1542)* status: all files reviewed, 4 unresolved discussions (waiting on @dorimedini-starkware, @elintul, and @MohammadNassar1)

___
*[`crates/blockifier/feature_contracts/cairo1/account_faulty.cairo` line 68 at r1](https://reviewable.io/reviews/starkware-libs/blockifier/1542#-Nqwv9Ld1xfKQz3dwfY8:-Nqwv9Ld1xfKQz3dwfY9:bjs5qu7) ([raw file](https://github.com/starkware-libs/blockifier/blob/8838261668fff28902a56a91d7a5eeb83889e1c1/crates/blockifier/feature_contracts/cairo1/account_faulty.cairo#L68)):*
> ```Cairo
>         let to_address = 0;
>         // By calling the `send_message_to_l1` function in validation and exeution, tests can now verify
>         // the functionality of entry point counters.
> ```

Which tests require this?



_Code quote:_
```Cairo
        // By calling the `send_message_to_l1` function in validation and exeution, tests can now verify
        // the functionality of entry point counters.
```
___
*[`crates/blockifier/feature_contracts/cairo1/account_faulty.cairo` line  at r1](https://reviewable.io/reviews/starkware-libs/blockifier/1542#-Nqwrut1Ap_O4Ajt0uJy:-Nqwu_9MAi6NgeReKXHQ:b-zccyoz) ([raw file](https://github.com/starkware-libs/blockifier/blob/78ac5d1365d5ec23c26c10a2472fa7e4cdb83034/crates/blockifier/feature_contracts/cairo1/account_faulty.cairo#L)):*
<details><summary><i>Previously, dorimedini-starkware wrote…</i></summary><blockquote>

there are many unrelated formatting changes here; why?
(you may be the only one with properly configured Cairo formatter lol)
</blockquote></details>

I can't do much about it, since I want to save my changes...

**Review by dorimedini-starkware — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/blockifier/1542)* status: all files reviewed, 3 unresolved discussions (waiting on @aner-starkware, @elintul, and @MohammadNassar1)

___
*[`crates/blockifier/feature_contracts/cairo1/account_faulty.cairo` line  at r1](https://reviewable.io/reviews/starkware-libs/blockifier/1542#-Nqwrut1Ap_O4Ajt0uJy:-Nqwvtza1OiT53suxwBo:blgh6s2) ([raw file](https://github.com/starkware-libs/blockifier/blob/78ac5d1365d5ec23c26c10a2472fa7e4cdb83034/crates/blockifier/feature_contracts/cairo1/account_faulty.cairo#L)):*
<details><summary><i>Previously, aner-starkware wrote…</i></summary><blockquote>

I can't do much about it, since I want to save my changes...
</blockquote></details>

It's fine :) my ctrl+S does nothing, so my theory still stands that you are the only dev that can format Cairo1 code automatically. own it

**Review by dorimedini-starkware — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/blockifier/1542)* status: all files reviewed, 3 unresolved discussions (waiting on @aner-starkware, @elintul, and @MohammadNassar1)

___
*[`crates/blockifier/feature_contracts/cairo1/account_faulty.cairo` line 68 at r1](https://reviewable.io/reviews/starkware-libs/blockifier/1542#-Nqwv9Ld1xfKQz3dwfY8:-NqwwaI59iD2yDJFCRiJ:b3go2nn) ([raw file](https://github.com/starkware-libs/blockifier/blob/8838261668fff28902a56a91d7a5eeb83889e1c1/crates/blockifier/feature_contracts/cairo1/account_faulty.cairo#L68)):*
<details><summary><i>Previously, aner-starkware wrote…</i></summary><blockquote>

Which tests require this?
</blockquote></details>

@MohammadNassar1 ?

**Comment by aner-starkware** [comment]
*[`crates/blockifier/feature_contracts/cairo1/account_faulty.cairo` line  at r1](https://reviewable.io/reviews/starkware-libs/blockifier/1542#-Nqwrut1Ap_O4Ajt0uJy:-NqwypwD3E3UqiSWmx2k:b8q9ans) ([raw file](https://github.com/starkware-libs/blockifier/blob/78ac5d1365d5ec23c26c10a2472fa7e4cdb83034/crates/blockifier/feature_contracts/cairo1/account_faulty.cairo#L)):*
<details><summary><i>Previously, dorimedini-starkware wrote…</i></summary><blockquote>

It's fine :) my ctrl+S does nothing, so my theory still stands that you are the only dev that can format Cairo1 code automatically. own it
</blockquote></details>

With the amount of errors VScode outputs when I open Cairo files, I'm not sure how much that is an advantage (or how reliable it is...)

**Review by MohammadNassar1 — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/blockifier/1542)* status: all files reviewed, 3 unresolved discussions (waiting on @aner-starkware, @dorimedini-starkware, and @elintul)

*[a discussion](https://reviewable.io/reviews/starkware-libs/blockifier/1542#-Nqwtg1F9EP6FoTBE2Yc:-Nqx4T5b8njR-9RPwbZ7:b4no98p) (no related file):*
<details><summary><i>Previously, dorimedini-starkware wrote…</i></summary><blockquote>

why were they added? which test needs them?
</blockquote></details>

please see my answer below.

___
___
*[`crates/blockifier/feature_contracts/cairo1/account_faulty.cairo` line 68 at r1](https://reviewable.io/reviews/starkware-libs/blockifier/1542#-Nqwv9Ld1xfKQz3dwfY8:-Nqx04Ri25Si6GgKUAgt:bzzg6q) ([raw file](https://github.com/starkware-libs/blockifier/blob/8838261668fff28902a56a91d7a5eeb83889e1c1/crates/blockifier/feature_contracts/cairo1/account_faulty.cairo#L68)):*
<details><summary><i>Previously, dorimedini-starkware wrote…</i></summary><blockquote>

@MohammadNassar1 ?
</blockquote></details>

Tests that use this account contract. For example, `test_validate_accounts_tx`.
We had a bug related to the entry points counter, which didn't consider `message_to_l1`.
By adding this syscall to `__validate__` and `__execute__`, we were able to reproduce the bug in our tests.

Bottom line: we need to keep these syscalls.

**Comment by aner-starkware** [comment]
*[`crates/blockifier/feature_contracts/cairo1/account_faulty.cairo` line 68 at r1](https://reviewable.io/reviews/starkware-libs/blockifier/1542#-Nqwv9Ld1xfKQz3dwfY8:-Nqx4qvk1-OG69xxBiW1:b-c4vnha) ([raw file](https://github.com/starkware-libs/blockifier/blob/8838261668fff28902a56a91d7a5eeb83889e1c1/crates/blockifier/feature_contracts/cairo1/account_faulty.cairo#L68)):*
<details><summary><i>Previously, MohammadNassar1 (mohammad-starkware) wrote…</i></summary><blockquote>

Tests that use this account contract. For example, `test_validate_accounts_tx`.
We had a bug related to the entry points counter, which didn't consider `message_to_l1`.
By adding this syscall to `__validate__` and `__execute__`, we were able to reproduce the bug in our tests.

Bottom line: we need to keep these syscalls.
</blockquote></details>

Then why don't any of these tests fail when these calls are removed?

**Review by MohammadNassar1 — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/blockifier/1542)* status: all files reviewed, 3 unresolved discussions (waiting on @aner-starkware, @dorimedini-starkware, and @elintul)

___
*[`crates/blockifier/feature_contracts/cairo1/account_faulty.cairo` line 68 at r1](https://reviewable.io/reviews/starkware-libs/blockifier/1542#-Nqwv9Ld1xfKQz3dwfY8:-Nqx6vwl9wBMG9CpdTJ6:b-ywg4tl) ([raw file](https://github.com/starkware-libs/blockifier/blob/8838261668fff28902a56a91d7a5eeb83889e1c1/crates/blockifier/feature_contracts/cairo1/account_faulty.cairo#L68)):*
<details><summary><i>Previously, aner-starkware wrote…</i></summary><blockquote>

Then why don't any of these tests fail when these calls are removed?
</blockquote></details>

It's part of the test. It tests the entry point counter,
By removing it, you are removing this check - that explains why tests still passing.
